### DISCUSSION TITLE
CumulativeConstraint

### DISCUSSION THREAD

**Discussion by erma07** [google/or-tools#3786] [Feature requests] [answered]
**What language and solver does this apply to?**
All, Python, Java, C#, C++ / CP-SAT

**Describe the problem you are trying to solve.**
CumulativeConstraint currently only ensures that for any integer point, the sum of the demands of the intervals containing that point does not exceed the capacity. I would like to have a constraint to achieve the opposite - for any integer point, the sum of the demands of the intervals containing that point exceeds the capacity
**Describe the solution you'd like**
for any integer point, the sum of the demands of the intervals containing that point exceeds the capacity
CumulativeConstraint AddCumulativeMin(LinearExpr capacity);

**Comment by lperron** [comment, marked as answer]
As answered on discord, this is an extremely complex code. So I will not code it.
The only solution is to count contributions on all time points, and make sure they are sufficient.

See http://www2.ift.ulaval.ca/~quimper/publications/min-cumulative.pdf
